frontend/src/app/modules/auth/components/review-report/review-report.component.ts

```diff
@@ -0,0 +1,183 @@
+        this.setAverageRatingStars(this.reviewReportResponse.averageRating);
+        this.toast.success("Review added")
+      }),
+      catchError(error => {
+        this.toast.error(error.error, 'Adding review failed');
+        throw error;
+      })
+    ).subscribe({
+      error: error => console.error('Error during adding review:', error)
+    });
+  }
+
+  onDeleteReview(reviewId: string, grade: number) {
+    console.log("DELETING REVIEW WITH ID " + reviewId)
+    this.deleteReview(reviewId, grade);
+  }
+
+  onUpdateReview(singleReview: UpdateSingleReview) {
+    this.updateReview(singleReview);
+  }
+
+  private deleteReview(reviewId: string, grade: number) {
+    this.deleteReviewSubscription = this.gradeService.deleteReview(reviewId).pipe(
+      tap(_ => {
+        this.setChangedFields(
+          this.reviewReportResponse.reviews.filter(review => review.id !== reviewId),
+          this.getNewAveragingRating(this.reviewReportResponse.averageRating, this.reviewReportResponse.totalReviews, grade, "MINUS"),
+          this.reviewReportResponse.totalReviews - 1,
+        );
+        this.setAverageRatingStars(this.reviewReportResponse.averageRating);
+        this.toast.success("Review deleted");
+      }),
+      catchError(error => {
+        this.toast.error(error.error, 'Deleting review failed');
+        throw error;
+      })
+    ).subscribe({
+      error: error => console.error('Error during deleting review:', error)
+    });
+  }
+
+  private updateReview(singleReview: UpdateSingleReview) {
+    const updateReviewRequest: UpdateReviewRequest = {
+      comment: singleReview.comment,
+      grade: singleReview.grade,
+      reviewType: 0,
+    }
+    this.updateReviewSubscription = this.gradeService.updateReview(singleReview.id, updateReviewRequest).pipe(
+      tap(_ => {
+        this.reviewReportResponse.reviews.forEach(review => {
+          if (review.id === singleReview.id) {
+            review.grade = singleReview.grade;
+            review.comment = singleReview.comment;
+            review.dateOfModification = new Date();
+          }
+        });
+        this.reviewReportResponse.averageRating = this.getNewAveragingRating(this.reviewReportResponse.averageRating, this.reviewReportResponse.totalReviews, singleReview.oldGrade, "UPDATE", singleReview.grade);
+        this.setAverageRatingStars(this.reviewReportResponse.averageRating);
+        this.toast.success("Review updated")
+      }),
+      catchError(error => {
+        this.toast.error(error.error, 'Updating review failed');
+        throw error;
+      })
+    ).subscribe({
+      error: error => console.error('Error during updating review:', error)
+    });
+  }
+
+  private getNewAveragingRating(averageRating: number, numberOfReviews: number, grade: number, op: string, updatedGrade?: number) {
+    if (numberOfReviews === 0){
+      if (op === "ADD") return grade;
```

Comment:
consider putting the ADD/UPDATE/.. strings into a constant for better maintainability and readability

Reply:
I've modified the code